### DISCUSSION TITLE
What's the best way to get chainId?

### DISCUSSION THREAD

**Discussion by bbzzzz** [wevm/wagmi#1085] [Q&A] [answered]
web3.js provides this [getChainId](https://web3js.readthedocs.io/en/v1.2.4/web3-eth.html#getchainid) method. What's the best way to get chainId using wagmi?

I noticed wagmi provides [useConnect](https://wagmi.sh/docs/hooks/useConnect) hook, and we can use the `onSuccess` callback to get chainId, but this only represents the chainId at the time of connection, user can change it later on, right?

I would like to get the chainId of the connected wallet, is this possible?

**Comment by tmm** [comment, marked as answer]
Check out [`useNetwork`](https://wagmi.sh/docs/hooks/useNetwork)

**Reply by kkagill** [reply]
@tmm hey, `activeChain` is undefined on initial page load. Any idea how to get value?
```
  const { chain: activeChain } = useNetwork();
  
  useEffect(() => {
    console.log(activeChain)  
  }, []);
```

**Reply by tmm** [reply]
You should add it to the dependency array as it can be `undefined` until a wallet is connected.
